aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -160,18 +160,21 @@ std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTab
   // For any dispatch key, it'll pick a kernel using the following order:
   //  (1) Use kernel if it's directly registered to this key
   //  (2) Handle runtime keys that have kernels available from alias keys
-  //    (2.1) Use kernel from DispatchKey::Autograd if available
-  //    (2.2) For autograd backend keys, we use kernel from alias Math key (catchAll will be moved to Math)
+  //    (2.1) Use kernel from DispatchKey::Math if available
+  //    (2.2) Use kernel from DispatchKey::Autograd if available
+  //    (2.3) Special logic to handle catchAll for Autograd keys
+  //          For autograd backend keys, we use kernel from alias Math key (catchAll will be moved to Math)
   //          if there's no direct registration to the backend key.
   //          Tensor factory functions used to have no registration to Autograd key but only to catchAll.
   //          In the past we directly call into backends(filled with catchAll) after BackendSelect.
   //          Now that we first call Autograd backend keys after BackendSelect, we should fill those
   //          with catchAll as well.
   //  (3) Use fallthrough kernel that are registered as fallback.
   //  (4) Use catchAll kernel if available
-  // TODO: currently Autograd is the only alias key, we'll update alias key precedence after we add new
-  //      alias keys AutogradDispatchCPUOrCUDA and Math.
-  // TODO: we can fix (2.2) and remove (4) after TypeDefault registrations are moved from catchAll to Math
+  // Alias Key Precedence:
+  //   Math > Autograd
```

Comment:
@bhosmer I think b) is also implicitly covered in the current logic where `Math>Autograd` + a)? 
When we have registration to both `Math` and `Autograd`:
1. If we don't have registration to backend X, for `AutogradX` it'll pick up `Math` kernel since `Math` has higher precedence. <= this is b? 
2. If we have registration to backend X, for `AutogradX` it'll skip 2.1 (due to `a`) and pick up `Autograd` kernel.

Reply:
Yeah I think that's right, in the case where we have both a Math kernel and an Autograd kernel. But the case I'm thinking of is the one where we have an Autograd kernel but no Math kernel (and no backend kernel) - in that case simple precedence says to plug in the Autograd kernel anyway, but I think we want to avoid that. 

(BTW I'm pretty sure at least one of our trial use cases has this Autograd-but-no-Math configuration - you can see how it's handled in the little working prototype.)

Reply:
We agreed offline when we don't have `Math` and backend `X` registered, `AutogradX` should pick up kernel from `Autograd` and error out when running inference. 
I've updated the colab notebook to match this behavior. :D

Reply:
Good. Now put this in the comments ;)